File: array.pas
program ArrayBasedList;

const
  MAX = 10;

type
  List = record
    data: array[1..MAX] of Integer;
    size: Integer;
  end;

// declare the variables
var
  L: List;
  i, pos, value: Integer;

procedure Initialize(var L: List);
begin
  L.size := 0;
end;

// insert
procedure Insert(var L: List; pos, value: Integer);
var
  i: Integer;
begin
  if L.size = MAX then
    writeln('List is full')
  else if (pos < 1) or (pos > L.size + 1) then
    writeln('Invalid position')
  else
  begin
    for i := L.size downto pos do
      L.data[i + 1] := L.data[i];
    L.data[pos] := value;
    L.size := L.size + 1;
  end;
end;

// delete
procedure Delete(var L: List; pos: Integer);
var
  i: Integer;
begin
  if (pos < 1) or (pos > L.size) then
    writeln('Invalid position')
  else
  begin
    for i := pos to L.size - 1 do
      L.data[i] := L.data[i + 1];
    L.size := L.size - 1;
  end;
end;

procedure Display(L: List);
var
  i: Integer;
begin
  if L.size = 0 then
    writeln('List is empty')
  else
    for i := 1 to L.size do
      write(L.data[i], ' ');
  writeln;
end;
// main program
begin
  Initialize(L);

  Insert(L, 1, 10);
  Insert(L, 2, 20);
  Insert(L, 3, 30);
  Display(L);

  Delete(L, 2);
  Display(L);
end.

File: array.java
public class array {
    static final int MAX = 10;

    static class List {
        int[] data = new int[MAX]; //stores the element
        int size;
    }

     // Initialize the list
    static void initialize(List L) {
        L.size = 0;//makes the list empty
    }

     // Insert at a given position
    static void insert(List L, int pos, int value) {

        if (L.size == MAX) {
            System.out.println("List is full");
            return;
        }

        if (pos < 0 || pos > L.size) {
            System.out.println("Invalid position");
            return;
        }

        // Shift elements to the right
        for (int i = L.size; i > pos; i--) {
            L.data[i] = L.data[i - 1];
        }
          L.data[pos] = value;
        L.size++;
    }

    // Delete an element from a given position
    static void delete(List L, int pos) {

        if (pos < 0 || pos >= L.size) {
            System.out.println("Invalid position");
            return;
        }

        // Shift elements to the left
        for (int i = pos; i < L.size - 1; i++) {
            L.data[i] = L.data[i + 1];
        }

        L.size--;
    }

    // Display the list
    static void display(List L) {

        if (L.size == 0) {
            System.out.println("List is empty");
            return;
        }

        for (int i = 0; i < L.size; i++) {
            System.out.print(L.data[i] + " ");
        }
        System.out.println();
    }
    public static void main(String[] args){
      List L = new List();
        initialize(L);

        insert(L, 0, 10);
        insert(L, 1, 20);
        insert(L, 2, 30);
        display(L);

        delete(L, 1);
        display(L);  
    }
}
